### DISCUSSION TITLE
Retrieving financial ratios, and financial data back in time (TTM & 5YA)

### DISCUSSION THREAD

**Discussion by mjhmilla** [alvarobartt/investpy#419] [Ideas]
Dear @alvarobartt 

I've been doing a survey of investing APIs and I am really impressed with what you've put together. As I am interested in fundamental data going as far back in time as possible, I would really like to be able to access the detailed information available from investing.com through your python API. For example, I'd love to be able to access every bit of data on the "ratios" page - such as the 'Return on Investment (TTM)', and 'Return on Investment (5YA)', etc. which you can see here:

https://www.investing.com/equities/siemens-ratios

Is this possible? To put it politely as possible, I'm quite willing to pay for some coffee, maybe for a lot of coffees, to have this feature added to the API.

With kind regards,

Matt

**Comment by datatalking** [comment]
I'm willing to look into this, what existing feature of investpy would this mimic or be similar to?

As far as data format, there are a few but i'm most familiar with pandas.

@mjhmilla what interface or setup do you currently have to trade ?

Any PR I submit would have to be approved by @alvarobartt
